packages/components/rating/src/Rating.tsx

```diff
@@ -1 +1,179 @@
+   */
+  onValueChange?: (value: number) => void
+  /**
+   * Sets the component as interactive or not.
+   * @default undefined
+   */
+  readOnly?: boolean
+  /**
+   * When `true`, prevents the user from interacting.
+   * @default false
+   */
+  disabled?: boolean
+  /**
+   * Sets the size of the stars.
+   * @default 'md'
+   */
+  size?: RatingStarProps['size']
+  /**
+   * Name of the underlying input.
+   * @default undefined
+   */
+  name?: string
+  /**
+   * id of the underlying input.
+   * @default undefined
+   */
+  id?: string
+  /**
+   * aria-label of the underlying input.
+   * @default undefined
+   */
+  'aria-label'?: string
+}
+
+export const Rating = forwardRef<HTMLDivElement, RatingProps>(
+  (
+    {
+      defaultValue,
+      value: propValue,
+      onValueChange,
+      size = 'md',
+      disabled,
+      readOnly,
+      name,
+      id,
+      ...rest
+    },
+    ref
+  ) => {
+    const inputRef = useRef<HTMLInputElement>(null)
+    const starRefList = useRef<HTMLDivElement[]>([])
+
+    const [value, setRatingValue] = useCombinedState(propValue, defaultValue, onValueChange)
+
+    const valueRef = useRef(value)
+    const isInteractive = !(disabled || readOnly)
+
+    function onStarClick(index: number) {
+      if (!inputRef.current) return
+
+      setRatingValue(index + 1)
+      valueRef.current = index + 1
+
+      inputRef.current.focus()
+      inputRef.current.setAttribute('data-clicked', '')
+    }
+
+    function onInputChange(event: ChangeEvent<HTMLInputElement>) {
+      // avoiding unnecessary calls to the onValueChange prop
+      // when the value remains unchanged
+      if (valueRef.current === Number(event.target.value)) return
+      valueRef.current = Number(event.target.value)
+
+      setRatingValue(Number(event.target.value))
+    }
+
+    function onStarMouseEnter({ currentTarget }: MouseEvent<HTMLDivElement>) {
+      const currentStarIndex = starRefList.current.findIndex(star => star === currentTarget)
+
+      const [previousStars, followingStars] = splitAt(starRefList.current, currentStarIndex + 1)
+
+      previousStars.forEach(star => star.setAttribute('data-hovered', ''))
+      followingStars.forEach(star => star.removeAttribute('data-hovered'))
+    }
+
+    const handleStarRef = useCallback((elm: HTMLDivElement | null) => {
+      if (!elm) return
+      starRefList.current.push(elm)
+    }, [])
+
+    function resetDataPartInputAttr() {
+      inputRef.current?.removeAttribute('data-clicked')
+    }
+
+    function resetDataPartStarAttr() {
+      starRefList.current.forEach(star => star.removeAttribute('data-hovered'))
+    }
+
+    return (
+      <div
+        className="relative"
+        ref={ref}
+        data-spark-component="rating"
+        {...rest}
+        onMouseLeave={resetDataPartStarAttr}
+      >
+        <input
+          name={name}
+          id={id}
+          aria-label={rest['aria-label'] ?? 'rating'}
```

Comment:
I wonder if we should do like IconButton and make the aria-label mandatory 🤔 

Here the risk is developers will never use it we should avoid to have hard-coded wordings in our components (this is part of our agreements).

For example `Dialog.CloseButton` does not have "close" as an aria-label by default.

WDYT ?

Reply:
Completely agree! @acd02 do you want to take the point?